cellrank/kernels/_moscot_kernel.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any, Dict, Type, Tuple, Union, Literal, Mapping, Optional, Sequence
+
+from types import MappingProxyType
+from moscot.base.output import BaseSolverOutput
+from moscot.base.problems import CompoundProblem
+
+import scanpy as sc
+import anndata
+from cellrank.kernels._transport_map_kernel import TransportMapKernel
+
+import numpy as np
+import scipy.sparse as sp
+
+__all__ = ["MoscotKernel"]
+
+
+class MoscotKernel(TransportMapKernel):
+    """Kernel for all moscot problems with a transport matrix as output."""
+
+    def __init__(
+        self, *args: Any, problem: Optional[Type[CompoundProblem]] = None, **kwargs: Any
+    ):
+        super().__init__(*args, **kwargs)
+        self._problem = problem
+
+    @classmethod
+    def load(
+        cls,
+        problem: Type[CompoundProblem],
+        backward: bool = False,
+        **kwargs: Any,
+    ) -> "MoscotKernel":
+        """Load from a moscot model."""
+        return MoscotKernel(
+            adata=problem.adata,
+            problem=problem,
+            time_key=problem.temporal_key,
+            backward=backward,
+            **kwargs,
+        )
+
+    def compute_transition_matrix(
```

Comment:
I don't think there's a need to override this, just use the `_compute_tmap` API which should provide you with everything.

Regarding the `tmaps` argument, I think this is generic enough that it should be in the `TransportKernel` class, as a class method, e.g.,:
```python
@classmethod
def from_transport_maps(cls, adata: AnnData, tmaps: Dict[..., ...], **kwargs: Any) -> "TransportMapKernel":
    # do some checking whether all tmaps are present and synced with `adata`, etc.
    kernel = cls(adata, ...) 
    kernel._tmaps = tmaps
    kernel._is_precomputed = True  # needs to be set to false in `TransportMapKernel.__init__`
    return kernel
```

and then in `compute_transition_matrix`, you'd have:
```python
if not self._is_precomputed:
    self._tmaps = {}
    for t1, t2 in tqdm(timepoints, unit="time pair"):
        tmat = self._compute_tmap(t1, t2, **kwargs)
        self.transport_maps[t1, t2] = self._tmat_to_adata(t1, t2, tmat)
```